server/src/main/java/org/apache/druid/segment/handoff/SegmentHandoffNotifierFactory.java

```diff
@@ -23,4 +23,5 @@
 public interface SegmentHandoffNotifierFactory
 {
   SegmentHandoffNotifier createSegmentHandoffNotifier(String dataSource);
+  SegmentHandoffNotifier createSegmentHandoffNotifier(String dataSource, String indexTaskId);
```

Comment:
You can just update the existing method rather than add a new one.

Also, use `taskId` instead of `indexTaskId`.

Reply:
Sure 👍